app/controllers/partners_controller.rb

```diff
@@ -18,9 +18,13 @@ def create
 
   def approve_application
     @partner = current_organization.partners.find(params[:id])
-    @partner.update(status: "Approved")
-    DiaperPartnerClient.put(@partner.attributes)
-    redirect_to partners_path
+    client_response = DiaperPartnerClient.put(@partner.attributes)
+    if client_response.is_a?(Net::HTTPSuccess)
+      @partner.update(status: "Approved")
```

Comment:
This is fine and doesn't need to be changed, but here's a cool Rails thing:

because `status` is an `enum` -- Rails, creates some cool magic methods. You can type the name of the status and end it with a bang (`!`) and it'll set that status for you. So in this case, it could be something like:
```suggestion
@partner.Approved!
```
(I'm not sure how case plays into this, so it *might* be `approved!` instead)

Reply:
I had to do a conflict resolve on this branch, and the other side had this enum setting magic method, so I brought that over. The rest of your code is as you wrote it though. :+1:

Reply:
Oh my gosh that is so cool! (Better late than never on writing back) That for sure makes things prettier!